internal/ingress/controller/parser/parser.go

```diff
@@ -942,7 +943,18 @@ func overrideRouteByKongIngress(route *Route,
 
 	r := kongIngress.Route
 	if len(r.Methods) != 0 {
-		route.Methods = cloneStringPointerSlice(r.Methods...)
+		invalid := false
+		for _, method := range r.Methods {
+			if !validMethods.MatchString(*method) {
+				// if any method is invalid (not an uppercase alpha string),
+				// discard everything
+				glog.Errorf("invalid method found while processing '%v': %v", route.Name, *method)
+				invalid = true
+			}
+		}
+		if !invalid {
+			route.Methods = cloneStringPointerSlice(r.Methods...)
+		}
```

Comment:
Nice work!
I like the way the failure is being processed here on a per value basis.
The only minor nit I can spot is the `route.Name` is constructed using Ingress resource and users can make out the name of namespace and Ingress resource. The best would be to output them properly. I can see that it might not be possible because you don't have access to those details inside this function.